x-pack/plugins/secops/server/graphql/scalar_date/resolvers.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { GraphQLScalarType, Kind } from 'graphql';
+
+export const dateScalar = new GraphQLScalarType({
+  name: 'Date',
+  description:
+    'Represents a Date for either an ES formatted date string or epoch string ISO8601 formatted',
+  serialize(value): string {
+    return Number.isNaN(Date.parse(value)) ? new Date(value).toISOString() : value;
```

Comment:
To make it more readable maybe we can use `moment` to validate if it is a date

Reply:
I would put this validator into a common utility with a few unit tests that just validate date times. We might need to pull in a special utility for it. I think we're going to have corner cases around whichever way we do it and we should do it consistently.

My 2 cents.

Reply:
> ~To make it more readable maybe we can use `moment` to validate if it is a date~

It seems like the isValid from `moment` is only validating ISO8601 formats, so it won't be really useful for what you are trying to do so forget about my suggestion,